openmmtools/mcmc.py

```diff
@@ -715,8 +717,10 @@ def apply(self, thermodynamic_state, sampler_state, context_cache=None):
                 # Run dynamics.
                 timer.start("{}: step({})".format(move_name, self.n_steps))
                 integrator.step(self.n_steps)
-            except Exception:
+            except Exception as e:
                 # Catches particle positions becoming nan during integration.
+                # Return the exception message as a warning
+                warnings.warn(str(e))
```

Comment:
This should do it, but we probably want to also add a message about how we will continue running

Reply:
Not sure what you mean by this exactly, the user does get an error message saying that it will be restarted, as per https://github.com/choderalab/openmmtools/pull/658/files#diff-85baf1df1041cffd52aca6ac6b2d4a7f00244716d1a0d14ac54af193c6835072R740-R742

Would this do it? I think we do want to refactor this restart logic to make it more readable and maintainable in the future.

Reply:
I'll be merging this for now, but feel free to raise an issue about this if needed.